src/Dict.cc

```diff
@@ -25,7 +25,7 @@
 
 namespace zeek {
 
-class IterCookie {
+class [[deprecated("Remove in v5.1. Use the standard-library-compatible version of iteration.")]] IterCookie {
```

Comment:
I was debating back and forth that this might land before 4.0, but it's probably best to hold onto it for after that since i have other changes to make still. I'll change them all to 5.1.

Reply:
I was debating holding onto this until after 4.0, since there's more work to do with the dictionary types until I feel like it's really "done". If you think we can go ahead and bring it in, then I'll fix them.

Reply:
The opening post of the PR had a thing about removing stuff within v4.1, so think I mistook the target of this to be merging before v4.0.  Waiting to find out what changes are needed to support the other pending improvements seems totally fine -- I'll change this to a draft and push milestone/project targets to v4.1.

Reply:
You're right, my mistake 👍